### DISCUSSION TITLE
Parse error: syntax error, unexpected ':', expecting '{' in /var/www/html/library/vendor/symfony/polyfill-php80/bootstrap.php on line 23

### DISCUSSION THREAD

**Discussion by Nyasha-13** [laravel/framework#34049] [General] [answered]
Local Machine
PHP Version 7.4.8
Laravel 7

Live Server 
PHP Version 7.2.31

I am getting this error when I move my laravel project to live server .What would be the problem and how can I get around it.

![image](https://user-images.githubusercontent.com/16807047/91544186-aafa2680-e91f-11ea-8292-4eaff55f16f4.png)

**Comment by driesvints** [comment]
Are you sure your live server is PHP 7.2? 

https://github.com/symfony/polyfill-php80/blob/master/bootstrap.php#L23

The error seems to hint at return types not being supported.

**Reply by Nyasha-13** [reply]
Yes my live server has PHP 7.2 ,Additionally I have another server running on PHP 7.3 which is also giving me the same error.

SOS!

**Comment by CloudyCity** [comment]
I have a similar problem when using composer or artisan command but no problem in cgi.

```
Local Machine
PHP Version 7.4.10
Laravel 7

Live Server
PHP Version 7.2.33
```

In my case,  i specify the php version in command and it work.
```
$ php7 artisan
$ php7 composer
```

Make sure your apache/nginx is listening the port of php7.

**Comment by Nyasha-13** [comment, marked as answer]
I finally fixed it by deploying the Laravel app on  fresh server with
- PHP version 7.3.22
 
I followed the procedures outlined here to install the environment >>
PHP Install
 [https://www.cloudbooklet.com/how-to-install-php-7-3-on-ubuntu-18-04/]

Also I installed Apache and mysql 

Then configured server to enable .htaccess by adding this to my vhost:
Edited 000-default.conf 

`sudo nano /etc/apache2/sites-available/000-default.conf`

Added the following on 000-default.conf whereby public is my public folder in my laravel app 

```
<Directory /var/www/laravel_app_folder_name/public/>
    Options Indexes FollowSymLinks
    AllowOverride All
    Require all granted
</Directory>
```

Then have mod_rewrite enabled.

`sudo a2enmod rewrite`

Finally restart apache service;

`sudo service apache2 restart`

I hope it helps!

**Comment by webfuelcode** [comment]
Why this happens. I uploaded laravel project on the server where PHP is 7.3

I change it to 7.4 and back to 7.3 and it runs. No matter it is PHP 7.3 or 7.4 but the first setup shows the error and changing and back to the same fixes it...?

**Comment by exdon10** [comment]
Your laravel version not compatable with your  Server
Use `composer install --ignore-platform-reqs`
`--ignore-platform-reqs`  ignores laravel versoin.
